common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -385,50 +388,53 @@ def _capture_basic_metrics(self):
         """
         Capture basic information about this sequence in New Relic.
         """
-        newrelic.agent.add_custom_parameter('seq.block_id', unicode(self.location))
-        newrelic.agent.add_custom_parameter('seq.display_name', self.display_name or '')
-        newrelic.agent.add_custom_parameter('seq.position', self.position)
-        newrelic.agent.add_custom_parameter('seq.is_time_limited', self.is_time_limited)
+        if newrelic:
+            newrelic.agent.add_custom_parameter('seq.block_id', unicode(self.location))
+            newrelic.agent.add_custom_parameter('seq.display_name', self.display_name or '')
+            newrelic.agent.add_custom_parameter('seq.position', self.position)
+            newrelic.agent.add_custom_parameter('seq.is_time_limited', self.is_time_limited)
 
     def _capture_full_seq_item_metrics(self, display_items):
         """
         Capture information about the number and types of XBlock content in
         the sequence as a whole. We send this information to New Relic so that
         we can do better performance analysis of courseware.
         """
-        # Basic count of the number of Units (a.k.a. VerticalBlocks) we have in
-        # this learning sequence
-        newrelic.agent.add_custom_parameter('seq.num_units', len(display_items))
+        if newrelic:
```

Comment:
I agree with @nedbat about the potential for another layer of abstraction, for the duplicate code in `if...else`, and for weird conditionals like this within helper methods.